webhook.go

```diff
@@ -208,6 +218,18 @@ func hookHandler(w http.ResponseWriter, r *http.Request) {
 	if matchedHook := matchLoadedHook(id); matchedHook != nil {
 		log.Printf("%s got matched\n", id)
 
+		// check if we have concurrency limits
+		if _, ok := limits[id]; ok {
+			if len(limits[id]) == cap(limits[id]) {
+				log.Printf("reached concurrency limit for: %s (max=%d)", id, len(limits[id]))
+				w.WriteHeader(http.StatusTooManyRequests)
+				fmt.Fprintf(w, "Error occurred while evaluating hook rules.")
```

Comment:
I would change the error message to something along these lines: `Hook reached maximum concurrent execution limit. Try again later.`

Reply:
fixed